jax/interpreters/xla.py

```diff
@@ -618,17 +733,34 @@ def _get_device(device, backend):
 xla_call_p.def_impl(_xla_call_impl)
 pe.staged_out_calls.add(xla_call_p)
 
-def _xla_call_translation_rule(c, axis_env,
-                               in_nodes, name_stack, backend, name,
-                               call_jaxpr, device=None):
+def _xla_call_translation_rule(
+    c: xc.XlaBuilder,
+    axis_env,
+    in_nodes,
+    name_stack: str,
+    backend,
+    name: str,
+    call_jaxpr,
+    device: Optional[xe.Device],
+    donate: Sequence[bool],
+):
   del device  # Ignored.
+  if not all(not x for x in donate):
+    raise ValueError("Donating buffers passed to a jit nested inside a jit or "
+                     "pmap is not supported.")
+
   subc = xb.make_computation_builder(f"jit_{name}")
   args = [xb.parameter(subc, i, c.GetShape(n)) for i, n in enumerate(in_nodes)]
   out_nodes = jaxpr_subcomp(subc, call_jaxpr, backend, axis_env, (),
                             extend_name_stack(name_stack, wrap_name(name, 'jit')), *args)
   subc = subc.Build(xops.Tuple(subc, out_nodes))
   return xops.Call(c, subc, list(in_nodes))
-ad.primitive_transposes[xla_call_p] = partial(ad.call_transpose, xla_call_p)
+
+def _xla_call_transpose_rule(params, call_jaxpr, args, ct):
+  params["donate"] = (False,) * len(args)
```

Comment:
This is suspicious, why should we let `donate` get out of sync with args?

Reply:
I was thinking that transpose of `f(x, y) -> z` is `f'(z) -> dx, dy`? Perhaps I've totally misunderstood?

Reply:
See above comments wrt JVPTrace, we are now handling this correctly where inputs to call primitives are changed and I've removed this.